src/libraries/Microsoft.NET.WebAssembly.Webcil/src/Webcil/WebcilWasmWrapper.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Collections.Immutable;
+using System.Reflection.PortableExecutable;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.WebAssembly.Webcil;
+
+//
+// This is a pretty dumb wrapper.  It assumes that the entire wasm module is going to be unchanging,
+// except for the data section which has 2 passive segments.  segment 0 is 4 bytes and contains the
+// length of the webcil payload.  segment 1 is of a variable size and contains the webcil payload.
+//
+// the unchanging parts are stored as a "prefix" and "suffix" which contain the bytes for the following
+// WAT program, split into the parts that come before the data section, and the bytes that come after:
```

Comment:
nit: i think in this case it's a WAT module; it's not a program since it isn't actually executable in a meaningful sense